src/app/shared/state/ontology/ontology.state.ts

```diff
@@ -11,27 +37,88 @@ import { OntologyNode, OntologyStateModel } from './ontology.model';
   defaults: {
     root: undefined,
     ids: [],
-    nodes: { }
+    nodes: {}
   }
 })
 export class OntologyState implements NgxsOnInit {
+  private ontologyUrl = environment.ontologyUrl;
+
   /**
    * Selects all nodes in the ontology tree.
    *
-   * @param state The ontology tree state.
-   * @returns All ontology nodes.
+   * @param state the ontology tree state.
+   * @returns all ontology nodes.
    */
   @Selector()
   static nodes(state: OntologyStateModel): OntologyNode[] {
-    return loValues(state.nodes);
+    return values(state.nodes);
   }
 
+  /**
+   *
+   * @param http instance of HttpClient
+   */
+  constructor(private http: HttpClient) { }
+
   /**
    * Ngxs' OnInit hook.
    *
    * @param ctx The state context.
    */
-  ngxsOnInit(ctx: StateContext<OntologyStateModel>) {
-    // TODO load ontology
+  ngxsOnInit(ctx: StateContext<OntologyStateModel>) { }
+
+  /**
+   * Fetches ontology as a json object and maps it to an ontology state model
+   * @returns ontology state model
+   */
+  private getOntology(): Observable<OntologyStateModel> {
+    return this.http.get(this.ontologyUrl, { responseType: 'text' }).pipe(
```

Comment:
Set response type to json (or remove the response type completely as json is the default type). You don't need to do JSON.parse at all.

Reply:
Done